bigquery_etl/glam/scalar_aggregates_incremental.py

```diff
@@ -1,68 +1,221 @@
+    destination_table,
+    attributes,
+    attributes_type,
+    user_data_type,
+    partition_clause,
+    **kwargs,
+) -> str:
+    """Render the table initialization DML for partitioning and clustering."""
+    env = Environment(loader=PackageLoader("bigquery_etl", "glam/templates"))
+    init_sql = env.get_template("clients_scalar_aggregates_v1.init.sql")
+    return reformat(
+        init_sql.render(
+            header=header,
+            destination_table=destination_table,
+            attributes_type=",".join(
+                f"{name} {dtype}" for name, dtype in zip(attributes, attributes_type)
+            ),
+            user_data_type=user_data_type,
+            partition_clause=partition_clause,
+        )
+    )
+
+
+def telemetry_variables():
+    """Telemetry specific variables."""
+    return dict(
+        user_data_type="""
+            ARRAY<
+                STRUCT<
+                metric STRING,
+                metric_type STRING,
+                key STRING,
+                process STRING,
+                agg_type STRING,
+                value FLOAT64
+                >
+            >
+        """,
+        user_data_attributes=[
+            "metric",
+            "metric_type",
+            "key",
+            "process",
+            # [agg_type, value] are shared between telemetry and glean
+        ],
+        attributes=["client_id", "os", "app_version", "app_build_id", "channel"],
+        attributes_type=["STRING", "STRING", "INT64", "STRING", "STRING"],
+        extract_select_clause=f"""
+            * EXCEPT(app_version),
+            CAST(app_version AS INT64) as app_version
+        """,
+        join_filter="""
+            LEFT JOIN
+                latest_versions
+            USING
+                (channel)
+            WHERE
+                app_version >= (latest_version - 2)
+        """,
+        partition_clause="""
+            PARTITION BY
+                RANGE_BUCKET(app_version, GENERATE_ARRAY(30, 200, 1))
+        """,
+    )
+
+
+def glean_variables():
+    """Glean/Fenix specific variables."""
+    return dict(
+        # does not include "process" field
+        user_data_type="""
+            ARRAY<
+                STRUCT<
+                metric STRING,
+                metric_type STRING,
+                key STRING,
+                agg_type STRING,
+                value FLOAT64
+                >
+            >
+        """,
+        user_data_attributes=[
+            "metric",
+            "metric_type",
+            "key",
+            # [agg_type, value] are shared between telemetry and glean
+        ],
+        attributes=[
+            "client_id",
+            "ping_type",
+            "os",
+            "app_version",
+            "app_build_id",
+            "channel",
+        ],
+        attributes_type=["STRING", "STRING", "STRING", "INT64", "STRING", "STRING"],
+        extract_select_clause="*",
+        # no filtering on channel
+        join_filter="",
```

Comment:
What is the reason for having an empty join filter here?

Reply:
I wasn't sure that the versioning scheme is the same, since Fenix doesn't follow the release train for Firefox. But, it seems okay to just look at the last two versions based on how long 2/3 have been around based on some poking around.

Reply:
If it looks reasonable to you to treat Fenix versioning in the same way, I think it makes for a good filter.

Reply:
Okay, I've added the query for `fenix_latest_versions_v1`. Ironically, this is the most expensive query out of all of them, coming in at 2GB.